shared/templates/mount_option_home/bash.template

```diff
@@ -0,0 +1,34 @@
+# platform = multi_platform_all
+# reboot = false
+
+{{% set non_allowed_partitions = ["/",
+                                  "/lib",
+                                  "/opt",
+                                  "/usr",
+                                  "/bin",
+                                  "/sbin",
+                                  "/boot",
+                                  "/dev",
+                                  "/proc"] %}}
+
+function perform_remediation (){
+
+    {{{ bash_ensure_mount_option_in_fstab( "$1", MOUNTOPTION, FILESYSTEM, TYPE ) | indent(4) }}}
+
+    {{{ bash_ensure_partition_is_mounted( "$1" ) | indent(4) }}}
+}
+
+readarray -t home_directories  < \
+    <(awk -F':' '{if ($3>=1000 && $3!= 65534) print $6}' /etc/passwd )
+
+
+for home_directory in "${home_directories[@]}"
+do
+    if [ -d $home_directory ]; then
```

Comment:
Well kind of... because in the case that the folder doesn't exists, how would the script know where should it be mounted? Or if for some reason the directory was deleted but the info in passwd wasn't updated, maybe this partition could not even exist.

I think this is the safest way to ensure we are setting the option to the right partition

Reply:
I agree the proper partition should be ensured, but my concern is about the limitations of how this is ensured now, relying on the home dir existence. There are valid cases where the home dir folder may not present and the partition still needs to be remediated. Could be simply because the home dir was not created or was deleted. But could also be cases where `automount` is dynamically mounting the partition, etc.

The logic should be adjusted to not depend on the user home dir existence to configure the partition options. Maybe the approach of using the `df` could be changed. There is also a fragility with the `df` approach, as I mentioned in the other comment.

The logic could be something like this:
- Collect all home dirs from interactive users from /etc/passwd, as already done.
- Collect the mounted partitions.
- Check if any home dir is below any mounted partition.
- If so, apply the remediation in the partition, as already done.

If this logic is too complex and not feasible for the moment, at least include a warning in these rules should be included, informing these limitations when this new template is used.

Reply:
There is an issue with your suggestion
If the right partition is below a wrong partition, there is the possibility to modify that wrong partition.
Taking the example in one of the first comments:
Let say we have a partition mounted in `/var` and there is another partition usually mounted in `/var/tmp` but not mounted now.
If there is an user with home directory in `/var/tmp/home_user1`, there is no way to ensure that the script will always find the partition usually mounted in `/var/tmp` and it then will modify the partition in `/var`

I'll set the warning for now